File: leetcode/560.py
class Solution:
  def subarraySum(self, nums:List[int], k:int) -> int:
    t, s = 0, 0
    sPrefix = {0: 1}
    for i in range(len(nums)):
      s += nums[i]
      t += sPrefix.get(s-k, 0)
      if not sPrefix.get(s): sPrefix[s] = 1
      else: sPrefix[s] += 1
    return t

File: leetcode/560.cpp
#include "../utils.h"

class Solution {
public:
  int subarraySum(vector<int>& nums, int k) {
    int n = nums.size();
    vector<int> pSum(n, 0);
    int s = 0;
    for(int i = 0; i < n; i++) {
      pSum[i] = nums[i]+s;
      s += nums[i];
    }
    printVector(pSum);
    return 0;
  }
};

int main() {
  Solution solve;
  vector<int> nums = {1,1,1};
  solve.subarraySum(nums, 3);
  nums = {1,2,3};
  solve.subarraySum(nums, 3);
  nums = {1,1,1,1,5,-5,2};
  solve.subarraySum(nums, 0);
  return 0;
}
